rasa/core/evaluation/marker_tracker_loader.py

```diff
@@ -0,0 +1,72 @@
+import random
+from rasa.shared.core.trackers import DialogueStateTracker
+from typing import Any, Iterable, List, Text, Optional
+from rasa.core.tracker_store import TrackerStore
+
+
+class MarkerTrackerLoader:
+    """Represents a wrapper around a `TrackerStore` that loads trackers in a configurable access pattern."""
+
+    @staticmethod
+    def strategy_all(count: int, keys: Iterable[Text]) -> Iterable[Text]:
+        """Selects all keys from the set of keys."""
+        return keys
+
+    @staticmethod
+    def strategy_first_n(count: int, keys: Iterable[Text]) -> Iterable[Text]:
+        """Takes the first N keys from the set of keys."""
+        return keys[:count]
+
+    @staticmethod
+    def strategy_sample(count: int, keys: Iterable[Text]) -> Iterable[Text]:
+        """Takes a sample of N keys from the set of keys."""
+        return random.choices(keys, k=count)
+
+    _STRATEGY_MAP = {
+        "all": strategy_all,
+        "first_n": strategy_first_n,
+        "sample": strategy_sample,
+    }
+
+    def __init__(
+        self,
+        tracker_store: TrackerStore,
+        strategy: str,
+        count: int = None,
+        seed: Any = None,
+    ) -> None:
+        """Create a MarkerTrackerLoader.
+        
+        Args:
+            tracker_store: The underlying tracker store to access.
+            strategy: The strategy to use for selecting trackers, can be 'all', 'sample', or 'first_n'.
+            count: Number of trackers to return, can only be None if strategy is 'all'.
+            seed: Optional seed to set up random number generator, only useful if strategy is 'sample'.
+        """
+        self.tracker_store = tracker_store
+        self.count = count
+
+        if strategy not in MarkerTrackerLoader._STRATEGY_MAP:
+            raise  # Not sure what best exception type to use here is, subclass of RasaException?
+
+        self.strategy = MarkerTrackerLoader._STRATEGY_MAP[strategy]
+
+        if not count:
+            if strategy != "all":
+                raise  # Exception
+            else:
+                pass  # Raise warning - redundant parameter
+
+        if seed:
+            if strategy == "sample":
+                random.seed(seed)
```

Comment:
I was thinking about this and I'm not sure this can guarantee us determinism, and I'm not 100% sure we can guarantee it depending on the backing store (basically, if the keys are returned from the backing store without consistent order then deterministically generating indices into that list does nothing) - but this might help a user make it deterministic if they account for that

Reply:
I think we can at least guarantee determinism if we're loading trackers from a folder (do we ever do that, or is it always from a tracker store?). 

I was under the impression that seeds should be set globally rather than inside the class, but can see that `RasaModel` [implements it the way you do](https://github.com/RasaHQ/rasa/blob/22751d7c60b8603c7d052d64060dc8eb5ff9bbab/rasa/utils/tensorflow/models.py#L90), so this will probably work. 

I'm not 100% sure how the seed interacts with the parallelisation you mentioned below. Do you know if it will work?

Reply:
As far as I'm aware we have to use a tracker store, and we can't just read files on disk - not seen any native capability to export trackers (except over the event broker which is a different kettle of fish).

That does set a global seed which is why I'm a bit skittish about it (especially when it concerns things that might not happen in a fixed order e.g. parallel execution). It might be slightly better to set the seed immediately before sampling but I think since we're parallelising _after_ sampling it's probably fine? Definitely something we'd need to test if we ever actually added a way of processing them in parallel

Reply:
> That does set a global seed which is why I'm a bit skittish about it (especially when it concerns things that might not happen in a fixed order e.g. parallel execution)

We can look into this a little bit more if we parallelise the execution. I'm sure we can control the behaviour of the seed.